diff --git a/src/compiler/nir/nir_print.c.org a/src/compiler/nir/nir_print.c
index a7ea12f..8ab4113 100644
--- a/src/compiler/nir/nir_print.c
+++ a/src/compiler/nir/nir_print.c
@@ -1710,6 +1710,7 @@ print_intrinsic_instr(nir_intrinsic_instr *instr, print_state *state)
          /* "General" handled above */
          nir_preamble_class cls = nir_intrinsic_preamble_class(instr);
          assert(cls == nir_preamble_class_image);
+         (void)cls;
          fprintf(fp, "class=image");
          break;
       }
