diff --git src/vulkan/util/meson.build src/vulkan/util/meson.build
index 90ebfd6..9a6063d 100644
--- src/vulkan/util/meson.build
+++ src/vulkan/util/meson.build
@@ -120,7 +120,7 @@ vk_extensions = custom_target(
 
 libvulkan_util = static_library(
   'vulkan_util',
-  [files_vulkan_util, vk_dispatch_table, vk_enum_to_str,
+  [files_vulkan_util, vk_enum_to_str,
    vk_struct_type_cast, vk_extensions],
   include_directories : [inc_include, inc_src],
   dependencies : [vulkan_wsi_deps, idep_mesautil, idep_nir_headers],
