diff --git meson.build meson.build
index 6b22930..4af29b1 100644
--- meson.build
+++ meson.build
@@ -582,7 +582,7 @@ if with_gallium_kmsro and (with_platform_x11 and not with_dri3)
 endif
 
 dep_dxheaders = null_dep
-if with_gallium_d3d12 or with_microsoft_clc or with_microsoft_vk
+if true or with_gallium_d3d12 or with_microsoft_clc or with_microsoft_vk
   dep_dxheaders = dependency('directx-headers', required : false)
   if not dep_dxheaders.found()
     dep_dxheaders = dependency('DirectX-Headers',
@@ -944,9 +944,6 @@ except:
 import mako
 assert Version(mako.__version__) >= Version("0.8.0")
   ''', check: false)
-if has_mako.returncode() != 0
-  error('Python (3.x) mako module >= 0.8.0 required to build mesa.')
-endif
 
 if cc.get_id() == 'gcc' and cc.version().version_compare('< 4.4.6')
   error('When using GCC, version 4.4.6 or later is required.')
@@ -1084,7 +1081,6 @@ if cc.get_argument_syntax() == 'msvc'
     '/wd5105',  # macro expansion producing 'defined' has undefined behavior (winbase.h, need Windows SDK upgrade)
     '/we4020',  # Error when passing the wrong number of parameters
     '/we4024',  # Error when passing different type of parameter
-    '/we4189',  # 'identifier' : local variable is initialized but not referenced
     '/Zc:__cplusplus', #Set __cplusplus macro to match the /std:c++<version> on the command line
   ]
   c_args += cc.get_supported_arguments(_trial)
@@ -1519,7 +1515,7 @@ if cc.links('int main() { return 0; }',
             args : '-Wl,--dynamic-list=@0@'.format(
               join_paths(meson.current_source_dir(), 'build-support/conftest.dyn')),
             name : 'dynamic-list')
-  with_ld_dynamic_list = true
+  with_ld_dynamic_list = false
 endif
 
 ld_args_build_id = cc.get_supported_link_arguments('-Wl,--build-id=sha1')
@@ -1716,7 +1712,7 @@ if dep_libudev.found()
   pre_args += '-DHAVE_LIBUDEV'
 endif
 
-llvm_modules = ['bitwriter', 'engine', 'mcdisassembler', 'mcjit', 'core', 'executionengine', 'scalaropts', 'transformutils', 'instcombine']
+llvm_modules = ['bitwriter', 'engine', 'mcdisassembler', 'mcjit', 'core', 'executionengine', 'scalaropts', 'transformutils', 'instcombine', 'passes']
 llvm_optional_modules = ['coroutines']
 if with_amd_vk or with_gallium_radeonsi or with_gallium_r600
   llvm_modules += ['amdgpu', 'bitreader', 'ipo']
diff --git src/gallium/drivers/llvmpipe/lp_tex_sample.h src/gallium/drivers/llvmpipe/lp_tex_sample.h
index 9e8baa0..e04f98a 100644
--- src/gallium/drivers/llvmpipe/lp_tex_sample.h
+++ src/gallium/drivers/llvmpipe/lp_tex_sample.h
@@ -36,7 +36,7 @@ struct lp_sampler_static_state;
 /**
  * Whether texture cache is used for s3tc textures.
  */
-#define LP_USE_TEXTURE_CACHE 0
+#define LP_USE_TEXTURE_CACHE 1
 
 struct lp_build_sampler_soa *
 lp_llvm_sampler_soa_create(const struct lp_sampler_static_state *static_state,
diff --git src/gallium/targets/libgl-gdi/meson.build src/gallium/targets/libgl-gdi/meson.build
index 7e1b87b..92f786a 100644
--- src/gallium/targets/libgl-gdi/meson.build
+++ src/gallium/targets/libgl-gdi/meson.build
@@ -36,8 +36,9 @@ libopengl32 = shared_library(
     inc_include, inc_wgl, inc_src, inc_gallium,
   ],
   link_with : [
-    libgallium_wgl, libglapi_bridge, libglapi
+    libgallium_wgl, libglapi_bridge, libglapi, libvulkan_util
   ],
+  link_whole : [ libgallium_wgl ],
   dependencies : [
     idep_mesautil
   ],
diff --git src/gallium/targets/wgl/meson.build src/gallium/targets/wgl/meson.build
index 2999ee5..8b92387 100644
--- src/gallium/targets/wgl/meson.build
+++ src/gallium/targets/wgl/meson.build
@@ -29,10 +29,9 @@ wgl_def = custom_target(
 )
 
 gallium_wgl_name = get_option('gallium-wgl-dll-name')
-libgallium_wgl = shared_library(
+libgallium_wgl = static_library(
   gallium_wgl_name,
   ['wgl.c'],
-  vs_module_defs : wgl_def,
   include_directories : [
     inc_include, inc_src, inc_mapi, inc_mesa, inc_gallium, inc_gallium_aux, inc_wgl, inc_gallium_winsys, inc_gallium_winsys_sw, inc_gallium_drivers,
   ],
diff --git src/meson.build src/meson.build
index ebfcf94..ce9855d 100644
--- src/meson.build
+++ src/meson.build
@@ -73,9 +73,7 @@ endif
 if with_platform_wayland
   subdir('egl/wayland/wayland-drm')
 endif
-if with_any_vk or with_gallium_zink
-  subdir('vulkan')
-endif
+subdir('vulkan')
 if with_any_intel
   subdir('intel')
 endif
diff --git src/util/futex.h src/util/futex.h
index 2a2a00c..5aff5ef 100644
--- src/util/futex.h
+++ src/util/futex.h
@@ -31,7 +31,7 @@
 #elif defined(__OpenBSD__)
 #define UTIL_FUTEX_SUPPORTED 1
 #elif defined(_WIN32) && !defined(WINDOWS_NO_FUTEX)
-#define UTIL_FUTEX_SUPPORTED 1
+#define UTIL_FUTEX_SUPPORTED 0
 #else
 #define UTIL_FUTEX_SUPPORTED 0
 #endif
